ts/environment.ts

```diff
@@ -3,59 +3,118 @@ import { hasTag, nixSource } from "./internal/utils.ts";
 import { Check } from "./check.ts";
 import { Executable } from "./executable.ts";
 import {
-  Interpolatable,
-  nixList,
+  NixStrLitInterpolatable,
   NixExpression,
+  nixList,
   nixRaw,
   nixStrLit,
 } from "./nix.ts";
 
+/**
+ * `Environment`s define what files and tools are available to `Executables`,
+ * `Check`s and development shells.
+ *
+ * For example they can contain compilers and developer tools that you want to
+ * use on a project, but they may also define other things like environment
+ * variables that need to be set.
+ *
+ * You can enter an `Environment` with `garn enter`.
+ */
 export type Environment = {
   tag: "environment";
   nixExpression: NixExpression;
+  /**
+   * Creates a new environment based on this one that includes the specified nix packages.
+   */
   withDevTools(devTools: Array<Package>): Environment;
-  shell(_s: TemplateStringsArray, ..._args: Array<Interpolatable>): Executable;
-  check(_s: TemplateStringsArray, ..._args: Array<Interpolatable>): Check;
+  /**
+   * Creates a new shell script `Executable`, run inside this `Environment`
+   */
+  shell(
+    _s: TemplateStringsArray,
+    ..._args: Array<NixStrLitInterpolatable>
+  ): Executable;
+  /**
+   * Creates a new shell script `Check`, run inside this `Environment`
+   */
+  check(
+    _s: TemplateStringsArray,
+    ..._args: Array<NixStrLitInterpolatable>
+  ): Check;
 };
 
-export const shell = (
+/**
+ * Creates a new shell script `Executable`, run in the `emptyEnvironment`.
+ */
+export function shell(
   s: TemplateStringsArray,
-  ...args: Array<Interpolatable>
-) => emptyEnvironment.shell(s, ...args);
+  ...args: Array<NixStrLitInterpolatable>
+) {
+  return emptyEnvironment.shell(s, ...args);
+}
 
-export const check = (
+/**
+ * Creates a new shell script `Check`, run in the `emptyEnvironment`.
+ *
+ * Example:
+ * ```typescript
+ * // Will fail if any `TODO`s are found in the build artifacts of myPkg:
+ * garn.check`! grep -r TODO ${myPkg}`;
+ * ```
+ */
```

Comment:
So the example here I think shows that you can check a package. Whether the package has source files or not is up to whatever created the package. That being said, maybe packgaes of source files are unidiomatic for garn. If this is the case I can update the example.

I think having a global check might still be nice since maybe I want to write a check that checks that a binary runs successfully in the empty environment

Reply:
Oh, I missed the `${myPkg}`. Yeah, LGTM for now then.